File: backend/schemas/discovery.py

from pydantic import BaseModel, EmailStr


class HostDiscoveryRequest(BaseModel):
    email: EmailStr


class HostEntry(BaseModel):
    hostname: str
    port: int
    protocol: str | None = None
    dns_record: str | None = None
    priority: int | None = None
    latency_ms: int | None = None
    status: str


class HostDiscoveryResponse(BaseModel):
    email: EmailStr
    completed: bool
    discovery_method: str
    results: list[HostEntry]

File: frontend/src/api/discovery.ts
import axios from '@/http/axios'
import { API } from './endpoints'

export interface HostDiscoveryRequest {
  email: string
}

export interface HostEntry {
  hostname: string
  port: number
  protocol?: string | null
  dns_record?: string | null
  priority?: number | null
  latency_ms?: number | null
  status: string
}

export interface HostDiscoveryResponse {
  email: string
  completed: boolean
  discovery_method: string
  results: HostEntry[]
}

/** Discover SMTP hosts for an email address */
export const discoverSmtpHosts = async (
  payload: HostDiscoveryRequest,
) => {
  const { data } = await axios.post<HostDiscoveryResponse>(
    API.smtpDiscovery,
    payload,
  )
  return data
}

/** Discover IMAP hosts for an email address */
export const discoverImapHosts = async (
  payload: HostDiscoveryRequest,
) => {
  const { data } = await axios.post<HostDiscoveryResponse>(
    '/discovery/imap-hosts',
    payload,
  )
  return data
}
